Package.swift

```diff
@@ -136,6 +136,18 @@ let package = Package(
             name: "TuistCacheIntegrationTests",
             dependencies: ["TuistCache", "TuistSupportTesting", "RxBlocking", "TuistCoreTesting"]
         ),
+        .target(
+            name: "TuistTemplate",
+            dependencies: ["SPMUtility", "TuistCore", "TuistSupport", "TuistLoader"]
```

Comment:
Following the [microfeatures](https://github.com/tuist/microfeatures-guidelines) approach, I'd not add dependencies between the features. If scaffolding using a template requires loading the template and then passing that down to the entity that scaffolds the code, I'd do that at the TuistKit level. Here's some pseudocode:

```swift
import TuistTemplate
import Loader

class ScaffoldCommand {
  func execute() throws {
    let template = try loadTemplate()
    try templateGenerator.generate(template)
  }
}
```

Reply:
I felt at the beginning that it'd be better to separate templates and project description as much as I could, but that makes no sense when templates are part of `ProjectDescription` now anyway. I'll do a better refactoring, I got a little bit astray with my implementation and there are some inconsistencies because of that.

Reply:
Moved `TemplateLoader` to `TuistLoader` and got rid of some code - which is always good 🙂